### PR TITLE
[fix] support linked @sveltejs/kit package in Vite 3.0.3+

### PR DESCRIPTION
**Pull request by benmccann**

### CODE REVIEW

**Review by Rich-Harris — APPROVED** [review]

**Review by Rich-Harris — APPROVED** [review]

**Review by Conduitry — CHANGES_REQUESTED** [review]
Vite 3.0.8 has been released now.

**Comment by Rich-Harris** [comment]
The reason the tests are failing is that [index.js](https://github.com/sveltejs/kit/blob/master/packages/kit/src/index/index.js) and [private.js](https://github.com/sveltejs/kit/blob/master/packages/kit/src/index/private.js) are now being imported twice, breaking the `instanceof` check. I'm not sure what changed in Vite to break this

**Comment by Rich-Harris** [comment]
It works with `3.0.2` and below, fails with `3.0.3` and above. Of the [changelog entries](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md#303-2022-07-25), the following commits look most suspicious:

* https://github.com/vitejs/vite/commit/d420f01
* https://github.com/vitejs/vite/commit/ea27701
* https://github.com/vitejs/vite/commit/c33e365

**Comment by Rich-Harris** [comment]
It appears to only be a problem when the `@sveltejs/kit` package is linked — i.e. it's affecting our tests, and any projects that have something like this...

```json
{
  "pnpm": {
    "overrides": {
      "@sveltejs/kit": "../kit/packages/kit",
```

...but not normal apps

File: .changeset/pink-gorillas-watch.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+'@sveltejs/adapter-cloudflare': patch
+'@sveltejs/adapter-cloudflare-workers': patch
+'@sveltejs/adapter-netlify': patch
+'@sveltejs/adapter-static': patch
+'@sveltejs/adapter-vercel': patch
+'@sveltejs/kit': patch
+---
+
+[chore] upgrade to Vite 3.0.8
```
Comment (benmccann) [inline .changeset/pink-gorillas-watch.md:10]:
```suggestion
[chore] upgrade to Vite 3.0.8 and esbuild 0.14.54
```

**Review by Rich-Harris — APPROVED** [review]

**Review by Conduitry — APPROVED** [review]

**Pull request merged**